source/main.cpp

```diff
@@ -1,318 +1,794 @@
 
-void savingsEnabled(ConfigItemBoolean *item, bool value)
-{
-    (void)item;
-    WUPS_StoreBool(nullptr, DST_ENABLED_CONFIG_ID, value);
-    enabledDST = value;
+    if (cfg::sync)
+        update_time(); // Update time when plugin is loaded.
 }
 
-void notifyEnabled(ConfigItemBoolean *item, bool value)
-{
-    (void)item;
-    WUPS_StoreBool(nullptr, NOTIFY_ENABLED_CONFIG_ID, value);
-    enabledNotify = value;
-}
 
-void onHourOffsetChanged(ConfigItemIntegerRange *item, int32_t offset)
+WUPS_GET_CONFIG()
 {
-    (void)item;
-    WUPS_StoreInt(nullptr, OFFSET_HOURS_CONFIG_ID, offset);
-    offsetHours = offset;
-}
-
-void onMinuteOffsetChanged(ConfigItemIntegerRange *item, int32_t offset)
-{
-    (void)item;
-    WUPS_StoreInt(nullptr, OFFSET_MINUTES_CONFIG_ID, offset);
-    offsetMinutes = offset;
-}
-
-WUPS_GET_CONFIG() {
-    if (WUPS_OpenStorage() != WUPS_STORAGE_ERROR_SUCCESS) {
+    if (WUPS_OpenStorage() != WUPS_STORAGE_ERROR_SUCCESS)
         return 0;
-    }
 
     WUPSConfigHandle settings;
-    WUPSConfig_CreateHandled(&settings, "Wii U Time Sync");
+    WUPSConfig_CreateHandled(&settings, PLUGIN_NAME);
 
     WUPSConfigCategoryHandle config;
     WUPSConfig_AddCategoryByNameHandled(settings, "Configuration", &config);
     WUPSConfigCategoryHandle preview;
     WUPSConfig_AddCategoryByNameHandled(settings, "Preview Time", &preview);
 
-    WUPSConfigItemBoolean_AddToCategoryHandled(settings, config, "enabledSync", "Syncing Enabled", enabledSync, &syncingEnabled);
-    WUPSConfigItemBoolean_AddToCategoryHandled(settings, config, "enabledDST", "Daylight Savings", enabledDST, &savingsEnabled);
-    WUPSConfigItemBoolean_AddToCategoryHandled(settings, config, "enabledNotify", "Receive Notifications", enabledNotify, &notifyEnabled);
-    WUPSConfigItemIntegerRange_AddToCategoryHandled(settings, config, "offsetHours", "Time Offset (hours)", offsetHours, -12, 14, &onHourOffsetChanged);
-    WUPSConfigItemIntegerRange_AddToCategoryHandled(settings, config, "offsetMinutes", "Time Offset (minutes)", offsetMinutes, 0, 59, &onMinuteOffsetChanged);
-
-    OSCalendarTime ct;
-    OSTicksToCalendarTime(OSGetTime(), &ct);
-    char timeString[256];
-    snprintf(timeString, 255, "Current Time: %04d-%02d-%02d %02d:%02d:%02d:%04d:%04d\n", ct.tm_year, ct.tm_mon + 1, ct.tm_mday, ct.tm_hour, ct.tm_min, ct.tm_sec, ct.tm_msec, ct.tm_usec);
-    WUPSConfigItemStub_AddToCategoryHandled(settings, preview, "time", timeString);
+    WUPSConfigItemBoolean_AddToCategoryHandled(settings, config, CFG_SYNC,
+                                               "Syncing Enabled",
+                                               cfg::sync,
+                                               [](ConfigItemBoolean*, bool value)
+                                               {
+                                                   WUPS_StoreBool(nullptr, CFG_NOTIFY, value);
+                                                   cfg::notify = value;
```

Comment:
I imagine you meant to store `CFG_SYNC` instead of `CFG_NOTIFY` here, right?

Reply:
Correct, that was a copy-paste error on my part.